File: step0130/domain-core/src/main/java/de/conciso/starter/PersonenService.java
package de.conciso.starter;

import java.util.Optional;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Slf4j
@Service
public class PersonenService implements Personen {

  private final PersonDAO personDAO;

  private final Auftraege auftraege;

  public PersonenService(PersonDAO personDAO, Auftraege auftraege) {
    this.personDAO = personDAO;
    this.auftraege = auftraege;
  }

  @Override
  public Person create(String vorname, String name) {
    var auftrag = auftraege.create(Auftrag.builder().bestellNummer(vorname).build());
    log.info("created auftrag: " + auftrag.getId());

    log.info("create person: {} {}", vorname, name);
    var person = new Person(vorname, name);
    return personDAO.save(person);
  }

  @Override
  public Optional<Person> findById(int id) {
    log.info("looking for person with id: " + id);
    var found = personDAO.findById(id);
    if (found.isEmpty()) {
      log.warn("no person found with id: " + id);
    }
    return found;
  }
}

File: python0100/personen_service.py
import logging

from models import Person
from personen import Personen


class PersonenService(Personen):
    _log = logging.getLogger(__qualname__)

    def __init__(self) -> None:
        self._store: dict[int, Person] = {}
        self._next_id: int = 1

    def create(self, vorname: str, name: str) -> Person:
        self._log.info("create person: %s %s", vorname, name)
        person = Person(id=self._next_id, vorname=vorname, name=name)
        self._store[self._next_id] = person
        self._next_id += 1
        return person

    def find_by_id(self, id: int) -> Person | None:
        self._log.info("looking for person with id: %d", id)
        found = self._store.get(id)
        if found is None:
            self._log.warning("no person found with id: %d", id)
        return found
